src/xenia/gpu/vulkan/vulkan_command_processor.cc

```diff
@@ -995,6 +1030,34 @@ bool VulkanCommandProcessor::IssueCopy() {
   int32_t dest_max_y = int32_t(
       (std::max(std::max(dest_points[1], dest_points[3]), dest_points[5])));
 
+  // Search recently used window scissor sizes for the actual dimensions of
+  // this copy (instead of the rounded-out tile size we currently have)
```

Comment:
Hmm... In IssueCopy, we're given the exact dimensions of the texture in one of the copy registers. Those dimensions are rounded up to the nearest tile size when fetching from the render cache.

Perhaps there is an issue with finding the correct image in the render cache?

Reply:
The RB_COPY_* registers don't contain the true image width, just the surface pitch (which is too large). The RB_COPY_DEST_PITCH register does contain the true height, so I should use that. Still, I've not found a register that contains the true width.

My understanding is that the copy command doesn't actually need the true (pixel-accurate) width because the GPU stores the texture in a padded/aligned format anyway, rounding up to the next block size, and that the cropping from surface pitch to actual width is handled at texture fetch time.

Reply:
So the EDRAM source information is stored in register `RB_SURFACE_INFO`, and the pitch of the image (in EDRAM tiles) is specified here.

The logical width and height of the destination texture is stored in RB_COPY_DEST_PITCH in pixels. This cannot be incorrect otherwise it would screw up later reads.

Is the game resolving to a texture of a certain size then aliasing that texture with a different size?

Reply:
I guess you could say that, yeah. It does some typical postprocessing blur operations where you sample a texture and render to an RT that's half the size of the input. It ends up needing some oddly-sized textures in that process.

Initial render: 150 x 100, resolved to 152x104
Pass 2: 75x50, resolved as 80x56
Pass 3: 37x25, resolved as 40x32
Pass 4: 18x12, resolved as 24x16
Pass 5: 9x6, resolved as 16x8
Pass 6: 4x3, resolved as 8x8
Pass 7: 2x1, resolved as 8x8

Reply:
Okay cool, your data is very useful - thanks.

Can you also see what texture size the game requests when attempting to sample the resolved textures?

Reply:
The first column of that data is both the window scissor size when rendering and the requested texture size when sampling -- they always match.